python/tvm/relay/testing/tflite.py

```diff
@@ -0,0 +1,468 @@
+        opcode: int
+            The operator's builtin_code
+        options_type: str
+            The operator's builtin_options_type
+        options: Dict[str, Any]
+            The operator's builtin_options
+        """
+        self.opcode = opcode
+        self.options_type = options_type
+        self.options = options
+        self.op_inputs_idx = []
+        self.op_outputs_idx = []
+
+
+def generate_tflite_model(
+    inputs: List[Tensor],
+    outputs: List[Tensor],
+    operator: Operator,
+) -> bytes:
+    """Generate a TensorFlow Lite model
+
+    Parameters
+    ----------
+    inputs: List[Tensor],
+        The list of input tensors
+    outputs: List[Tensor],
+        The list of output tensors
+    operator: Operator,
+        The operator in the model
+
+    Returns
+    ------------
+    TensorFlow Lite model as bytes
+    """
+    tmp_dir = tempfile.gettempdir()
+
+    schema_path = tmp_dir + "/schema.fbs"
+
+    download(SCHEMA_URL, schema_path)
+
+    json_path = tmp_dir + "/tflite_model.json"
+    tflite_model_path = tmp_dir + "/tflite_model.tflite"
+
+    # figure out which input tensors are inputs to the model and which are inputs to the op
+    model_inputs_idx = []
+
+    for idx, tensor in enumerate(inputs):
+        # all input tensors are inputs to the operator
+        operator.op_inputs_idx.append(idx)
+        if tensor.buffer_data is None:
+            model_inputs_idx.append(idx)
+
+    tensors = inputs + outputs
+    # model and operator has the same output tensors
+    model_outputs_idx = list(range(len(inputs), len(tensors)))
+    operator.op_outputs_idx = model_outputs_idx
+
+    model_json = _make_json(tensors, operator, model_inputs_idx, model_outputs_idx)
+    with open(json_path, "w") as json_file:
+        json_file.write(model_json)
+
+    subprocess.run(
+        ["flatc", "-b", schema_path, json_path],
+        cwd=tmp_dir,
+        check=True,
+    )
+
+    with open(tflite_model_path, "rb") as file:
+        model = file.read()
+    return model
+
+
+def _make_json(
+    tensors: List[int],
+    operator: Operator,
+    model_inputs_idx: List[int],
+    model_outputs_idx: List[int],
+) -> str:
+
+    # first element in list of buffers is always an empty list
+    buffers = [{"data": []}]
+
+    # turn the Tensor objects into JSONable dicts
+    tensors_as_json = []
+    for idx, tensor in enumerate(tensors, start=1):
+        tensor.buffer_idx = idx
+        tensor.name = "x-" + str(idx)
+        tensors_as_json.append(tensor.to_json())
+
+        buffers.append({"data": tensor.buffer_data if tensor.buffer_data else []})
+
+    op = {
+        "opcode_index": 0,
+        "inputs": operator.op_inputs_idx,
+        "outputs": operator.op_outputs_idx,
+        "mutating_variable_inputs": [],
+    }
+    if operator.options_type != "":
+        op["builtin_options_type"] = operator.options_type
+        op["builtin_options"] = operator.options
+
+    dictionary = {
+        "version": 3,
```

Comment:
Perhaps this 3 shouldn't be hard-coded here. I'm not entirely sure what it refers to, but if it's schema versioning maybe a global at the top of the file is more appropriate.

Reply:
Yeah it's a schema versioning, I'll add it as a global variable to the top.